src/sensai/vector_model.py

```diff
@@ -161,39 +163,61 @@ def isFitted(self) -> bool:
         pass
 
 
-class RuleBasedModel(FittableModel, ABC):
-    """
-    Base class for models where the essential prediction logic is based on rules coded by humans
-    and thus does not require fitting. However, the input generation process may use mechanisms
-    such as feature generation or data transformation, which may require fitting.
-    """
-    def __init__(self):
-        super().__init__()
+class ClassificationMixin(ABC):
```

Comment:
Had to do a mixin because RuleBasedClassificationModel cannot inherit from the corresponding vector model (for performance reasons). The mixin might also be useful in other situations, I think

Reply:
This is becoming overly complex. I shall make a proposal (via a PR) on how to simplify things.
Please don't make further changes to vector_model until I'm done.

Reply:
Awesome, thank you!